dask/dataframe/core.py

```diff
@@ -2745,6 +2745,41 @@ def _repr_html_(self):
         return self._HTML_FMT.format(data=data, name=key_split(self._name),
                                      task=len(self.dask))
 
+    def _select_columns_or_index(self, columns_or_index):
+        """
+        :param columns_or_index: Column name or list of column names
+        and/or the name of the DataFrame's index
+        :return: Dask DataFrame with columns corresponding to each column or
+        index level in columns_or_index.  If included, the column corresponding 
+        to the index level is named _index
+        """
+        # Ensure columns_or_index is a list
+        columns_or_index = (columns_or_index
+                            if isinstance(columns_or_index, list)
+                            else [columns_or_index])
+
+        column_names = [n for n in columns_or_index
```

Comment:
Does `self.columns & columns_or_index` achieve the desired result? I'm wondering about cases where `n` is not a scalar, and why those should be excluded from `column_names`.

Reply:
Yeah, good catch. The scalar check was there to filter out cases where you're grouping on something like a series.  I'll extract the scalar check into a method that handles tuples and is a bit clearer.